### PR TITLE
Add doc for debugging rest tests

### PR DESCRIPTION
**Pull request by bowenlan-amzn**
### Description

Add doc for how to debug the OpenSearch cluster when running the rest tests

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- ~~[ ] Functionality includes testing.~~
- ~~[ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.~~
- ~~[ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.~~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by mch2 — APPROVED** [review]

File: TESTING.md [hunk 1]
```diff
@@ -274,6 +274,14 @@ If in doubt about which command to use, simply run &lt;gradle path&gt;:check
 
 Note that the REST tests, like all the integration tests, can be run against an external cluster by specifying the `tests.cluster` property, which if present needs to contain a comma separated list of nodes to connect to (e.g. localhost:9300).
 
+## Debug REST tests
```
Comment (dblock) [inline TESTING.md:277]: Capitalize properly, "Debugging REST Tests"

File: TESTING.md [hunk 2]
```diff
@@ -274,6 +274,14 @@ If in doubt about which command to use, simply run &lt;gradle path&gt;:check
...
+
+Since Rest tests can be run against an external cluster. You can launch an OpenSearch cluster in debug mode following [Launching and debugging from an IDE](#launching-and-debugging-from-an-ide). And connect your Rest test to this cluster.
```
Comment (dblock) [inline TESTING.md:279]: The first sentence isn't complete, just combine it all: 

```
To run REST tests against an external cluster specify the following options.

...

You can also launch an OpenSearch cluster in debug mode following [Launching and debugging from an IDE](#launching-and-debugging-from-an-ide)
```.

**Review by dblock — CHANGES_REQUESTED** [review]
Let's make this a little clearer?

**Comment by bowenlan-amzn** [comment]
> Let's make this a little clearer and add to TOC please.

Thanks @dblock . I update the doc, pls take another look when you have time.

**Review by dblock — APPROVED** [review]

**Pull request merged**